### DISCUSSION TITLE
Far Manager integrated to Visual Studio Code

### DISCUSSION THREAD

**Discussion by valorisa** [FarGroup/FarManager#740] [Ideas]
Hello,

I'd like to share the idea that far manager could be integrated into Visual Studio Code.
Thanks.
Bertrand B.

**Comment by yegor-mialyk** [comment]
Are you seeking assistance on how to integrate Far Manager into VS Code? Or are you simply informing the community about the possibility of doing so?

**Reply by valorisa** [reply]
I'm seeking assistance.

**Reply by yegor-mialyk** [reply]
To incorporate Far Manager into your Visual Studio Code environment as a terminal option, edit your `settings.json` file to include:

```
  "terminal.integrated.profiles.windows": {
    "Far Manager": {
      "path": "C:\\Far\\Far.exe"
    }
  }
```

Once this is configured, you can launch a new terminal instance by pressing `Ctrl+~`. At the top-right of the Terminal pane, use the `+` button, which has a drop-down arrow, to choose Far Manager as the terminal. You can also set it as the default terminal through this menu.

**Reply by valorisa** [reply]
OK. Thanks. Exactly what i was searching.

**Comment by alabuzhev** [comment]
VS code, same as VS, has an integrated terminal.
You can run Far there and it will look and feel quite integrated.

**Reply by valorisa** [reply]
Thanks.

**Comment by nightroman** [comment]
For cases like mine, to use Far as command line on steroids with hidden panels and the current directory set to VSCode current:

```json
  "terminal.integrated.profiles.windows": {
    "Far": {
      "path": "C:\\Bin\\Far\\x64\\Far.exe",
      "args": ["-set:Panel.Left.Visible=false", "-set:Panel.Right.Visible=false", "-set:Panel.Left.Folder=.", "-set:Panel.Right.Folder=."]
    },
```

**Reply by yegor-mialyk** [reply]
Thank you. Just for the reference, sometimes you want to run the terminal inside your working folder, in this case you need to replace `.` with `${workspaceFolder}`. A list of all variables you can use is here: https://code.visualstudio.com/docs/editor/variables-reference

**Reply by nightroman** [reply]
`${workspaceFolder}` is definitely better than `.`
Especially because of this unexpected effect - https://github.com/FarGroup/FarManager/issues/746#issuecomment-1802579353

**Comment by Arkady23** [comment]
Wow... It's a good idea to be on the github. I love Far. I develop programs in C# FrameWork 4, F# FrameWork 7 only on the Far editor. There is not enough feature for highlighting lines with some bright background, such as sometimes happens with the input of the phrase TODO. It works, but not always. And by the way, I haven't found a documented description that describes how to use TODO to make everything more beautiful. Perhaps you can use, say, three asterisks to highlight a line in a bright color, yellow or purple.
Russky:
Вау... Хорошая идея быть на гитхабе. Люблю Far. Разрабатываю программы на C# FrameWork 4, F# FrameWork 7 только на Far-редакторе. Не хватает фитчи выделения строчек каким-нибудь ярким фоном типа как иногда получается с вводом фразы TODO. Получается, но не всегда. И кстати я не нашел задокументированное описание, где бы описывалось как надо использовать TODO чтобы всё было покрасивее. Возможно можно использовать скажем три звездочки для выделения строки ярким цветом, желтым или фиолетовым.
